src/Eshopworld.Messaging/TopicAdapter.cs

```diff
@@ -28,18 +31,23 @@ internal sealed class TopicAdapter<T> : ServiceBusAdapter<T>
         /// <param name="subscriptionId">The ID of the subscription where the service bus namespace lives.</param>
         /// <param name="messagesIn">The <see cref="IObserver{IMessage}"/> used to push received messages into the pipeline.</param>
         /// <param name="batchSize">The size of the batch when reading for a queue - used as the pre-fetch parameter of the </param>
-        /// <param name="typeOverride">The type override when we're creating a topic adapter for <see cref="string"/> types.</param>
-        public TopicAdapter([NotNull]string connectionString, [NotNull]string subscriptionId, [NotNull]IObserver<T> messagesIn, int batchSize, Type typeOverride = null)
-            : base(connectionString, subscriptionId, messagesIn, batchSize, typeOverride)
+        /// <param name="messenger">The <see cref="Messenger"/> instance that created this adapter.</param>
+        public TopicAdapter([NotNull]string connectionString, [NotNull]string subscriptionId, [NotNull]IObserver<T> messagesIn, int batchSize, Messenger messenger)
+            : base(messagesIn, batchSize)
         {
-            if (typeof(T) == typeof(Message) && typeOverride == null)
-                throw new InvalidOperationException($"You can't create a TopicAdapter of type {typeof(Message).FullName} without specifying a typeOverride");
+            ConnectionString = connectionString;
+            Messenger = messenger;
+
+            TopicType = typeof(T);
 
-            if(typeof(T) != typeof(Message) && typeOverride != null)
-                throw new InvalidOperationException($"typeOverride is only respected if you're creating a TopicAdapter where T:{typeof(Message).FullName}, and this one is for {typeof(T).FullName}");
+            if (TopicType.FullName?.Length > 260) // SB quota: Entity path max length
```

Comment:
in which case typeof(T).FullName might be null?

Reply:
It's not marked as `[NotNull]` so R# complains of possible nullability without checks. It's either:

- this
- R# ignore once comment
- Do nothing, but R# error in designer